### PR TITLE
Enable compactor and alertmanager in target all

### LINKED ISSUE #6098: Enable compactor and alertmanager in target all
**Describe the bug**
target all doesn't include compactors and alertmanager

**Expected behavior**
Compactor and alertmanager are kind of essential and should be included in target all

### PR DESCRIPTION
**Pull request by SungJin1212**
**What this PR does**:
Enable compactor and alertmanager in target all.
**Which issue(s) this PR fixes**:
Fixes #6098 

**Checklist**
- [x] Tests updated
- [ ] Documentation added
- [x] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

**Comment by SungJin1212** [comment]
@friedrichg 
Could you please take a look?

**Comment by SungJin1212** [comment]
@friedrichg 
I fixing the e2e test now!

File: pkg/cortex/cortex_test.go [hunk 1]
```diff
@@ -92,10 +94,23 @@ func TestCortex(t *testing.T) {
...
+		AlertmanagerStorage: alertstore.Config{
+			Config: bucket.Config{
+				Backend: "local",
+			},
+		},
+		Target: []string{All},
 	}
 
+	externalURL := flagext.URLValue{}
+	err := externalURL.Set("http://localhost/api/prom")
```
Comment (friedrichg) [inline pkg/cortex/cortex_test.go:106]: http://localhost/api/prom
```suggestion
	err := externalURL.Set("http://localhost/alertmanager")
```
it's more appropiate

**Comment by friedrichg** [comment]
I think you can fix most tests adding the following snippet to the yamls in docs/configuration
```
alertmanager:
  external_url: http://localhost/alertmanager

alertmanager_storage:
  backend: local
```
or something similar

**Review by friedrichg — APPROVED** [review]
You didn't do what I suggested, but that is ok. You solved the problem.
The getting-started is not used in the docs anymore, so more work is needed there. Not to be done in this PR, for sure.

**Comment by SungJin1212** [comment]
@CharlieTLe 
Could you please take a look?

**Review by CharlieTLe — APPROVED** [review]

**Review by yeya24 — APPROVED** [review]

**Comment by yeya24** [comment]
I think this change breaks something but I am not sure why it is not captured by tests.

For example, it failed to run Cortex with provided example config file https://github.com/cortexproject/cortex/blob/master/docs/configuration/single-process-config-blocks-local.yaml because the config file misses alertmanager configurations

**Comment by SungJin1212** [comment]
I added alertmanager storage configs in this way.
```
configOverrides := map[string]string{
		// alert manager
		"-alertmanager.web.external-url":   "http://localhost/alertmanager",
		"-alertmanager-storage.backend":    "local",
		"-alertmanager-storage.local.path": filepath.Join(e2e.ContainerSharedDir, "alertmanager_configs"),
}
	
require.NoError(t, copyFileToSharedDir(s, "docs/configuration/single-process-config-blocks-local.yaml", cortexConfigFile))

cortex1 := e2ecortex.NewSingleBinaryWithConfigFile("cortex-1", cortexConfigFile, configOverrides, "", 9009, 9095)
```

**Comment by yeya24** [comment]
I see. @SungJin1212 Can we please fix the config file as well?
I have a user reporting to me that they failed to use the existing config after upgrading Cortex

**Comment by SungJin1212** [comment]
@yeya24 
Sure!

**Comment by SungJin1212** [comment]
@yeya24 
I make a PR. #6274

**Pull request merged**